src/app/components/Scan/Scan.tsx

```diff
@@ -29,20 +28,18 @@ function Scan(): JSX.Element {
       <p className={styles.title}>ScanCam</p>
       {content}
       <ImageInput onUpload={setImageURL} />
-      {recognizedProgress ? (
-        <Progress
-          progress={recognizedProgress.progress * 100}
-          status={recognizedProgress.status}
-        />
+      {!text && progress && (
+        <Progress progress={progress.progress} status={progress.status} />
+      )}
+      {text ? (
+        <div>{text && <AddDocumentForm text={text} />}</div>
```

Comment:
To make your code more clearly, you could transfer these 2 conditions also to your content variable.